src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILImporter.Interpreter.cs

```diff
@@ -155,22 +155,22 @@ private void ImportReturn()
             switch (stackItem.Kind)
```

Comment:
This switch should be based on the actual return type of the method, not based on the stackItem type.

E.g. if the stackItem is `Int32` and the actual type of the method is `Int64`, we need to to set the return value as `Int64`.

You do not have to fix this in this PR. I am mentioning it because of this uses `stackItem.Type` for float case right now, but it does not really need to.